torch/cuda/amp/amp_scaler.py

```diff
@@ -0,0 +1,332 @@
+import torch
+from collections import defaultdict
+from torch._six import container_abcs
+
+
+class _PerDeviceHelper(object):
+    """
+    Helper class to serve copies of a master tensor that may be needed on multiple devices.
+    Intended to be used as a temporary.
```

Comment:
Clarify what you mean by this. It's that the replicated tensors are only to be used temporarily, right?

Reply:
Oh no, it seems like the logic is even more complicated than that (per below). 

Is there a better way to do this? Maybe with getters/setters on the scaler class?